### PR TITLE
Fix parsing truncated DNS messages

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

File: akka-actor-tests/src/test/scala/akka/io/dns/internal/MessageSpec.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+package akka.io.dns.internal
+
+import akka.util.ByteString
+import org.scalatest.{ Matchers, WordSpec }
+
+class MessageSpec extends WordSpec with Matchers {
+  "The Message" should {
+    "parse a response that is truncated mid-message" in {
+      val bytes = ByteString(0, 4, -125, -128, 0, 1, 0, 48, 0, 0, 0, 0, 4, 109, 97, 110, 121, 4, 98, 122, 122, 116, 3, 110, 101, 116, 0, 0, 28, 0, 1)
+      val msg = Message.parse(bytes)
```
Comment (chbatey) [inline akka-actor-tests/src/test/scala/akka/io/dns/internal/MessageSpec.scala]: just asserting that it didn't throw?

Comment (raboof) [reply akka-actor-tests/src/test/scala/akka/io/dns/internal/MessageSpec.scala]: Yes currently. I'll add some more checks :)

**Review by chbatey — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**